engine/src/witness/eth/sc_utils.rs

```diff
@@ -99,32 +78,38 @@ impl<Inner: ChunkedByVault> ChunkedByVaultBuilder<Inner> {
 				.await?
 				{
 					info!("Handling event: {event}");
-					// TODO: To decode the call in the SC instead.
-					let _call: state_chain_runtime::RuntimeCall = match event.event_parameters {
-						ScUtilsEvents::DepositToScGatewayAndScCallFilter(DepositToScGatewayAndScCallFilter {
-							sender, // eth_address to attribute the FLIP to
-                            signer, // `tx.origin``. Not to be used for now
-                            amount, // FLIP amount deposited
-                            sc_call
-						}) => {
-                            match ScCallViaGateway::decode(&mut &sc_call[..]) {
-                                Ok(ScCallViaGateway::DelegateTo { operator }) => {
-									println!("Successfully Decoded ScCall! {:?}, deposited: {amount} FLIP, sender: {sender}, signer: {signer}, epoch index {:?}", operator, epoch.index);
-                                    trace!("Successfully decoded ScCall::DelegateTo with operator: {operator}");
-                                },
-                                Err(_e) => {
-									println!("Failed to decode ScCall");
-                                }
-                            }
-                            continue
-                        },
+					let call: state_chain_runtime::RuntimeCall = match event.event_parameters {
+						ScUtilsEvents::DepositToScGatewayAndScCallFilter(
+							DepositToScGatewayAndScCallFilter {
+								sender,    // eth_address to attribute the FLIP to
+								signer: _, // `tx.origin``. Not to be used for now
+								amount,    // FLIP amount deposited
+								sc_call,
+							},
+						) => {
+							pallet_cf_funding::Call::execute_sc_call {
+								sc_call: sc_call.to_vec(),
+								deposit_and_call:
+									DepositAndSCCallViaEthereum::FlipToSCGatewayAndCall {
+										amount: amount.try_into().unwrap(),
+										call: None, /* This will be filled on the SC after SC
+										             * decodes
+										             * the call from the sc_call bytes above */
+									},
```

Comment:
I don't understand why we need to construct this struct with a `None` value here... we could just pass the deposit amount and construct the call on-chain after decoding the data?